plugins/native/module/vtkF3DQuakeMDLImporter.cxx

```diff
@@ -0,0 +1,536 @@
+  std::vector<int> ActiveAnimationIds;
+  int NumberOfAnimations = 0;
+  double FrameRate = 60.0;
+  vtkSmartPointer<vtkTexture> Texture;
+};
+
+//----------------------------------------------------------------------------
+vtkF3DQuakeMDLImporter::vtkF3DQuakeMDLImporter()
+  : Internals(new vtkF3DQuakeMDLImporter::vtkInternals(this))
+{
+}
+
+//----------------------------------------------------------------------------
+int vtkF3DQuakeMDLImporter::ImportBegin()
+{
+  return this->Internals->ReadScene(this->FileName);
+}
+
+//----------------------------------------------------------------------------
+void vtkF3DQuakeMDLImporter::ImportActors(vtkRenderer* renderer)
+{
+  this->Internals->ImportActors(renderer);
+}
+
+//----------------------------------------------------------------------------
+bool vtkF3DQuakeMDLImporter::UpdateAtTimeValue(double timeValue)
+{
+  this->Internals->UpdateTimeStep(timeValue);
+  return timeValue <= this->Internals->FrameRate * this->Internals->AnimationIds.size();
+}
+
+//----------------------------------------------------------------------------
+vtkIdType vtkF3DQuakeMDLImporter::GetNumberOfAnimations()
+{
+  return this->Internals->NumberOfAnimations;
+}
+
+//----------------------------------------------------------------------------
+std::string vtkF3DQuakeMDLImporter::GetAnimationName(vtkIdType animationIndex)
+{
+  assert(animationIndex < static_cast<vtkIdType>(this->Internals->AnimationNames.size()));
+  assert(animationIndex >= 0);
+  return this->Internals->AnimationNames[animationIndex];
+}
+
+//----------------------------------------------------------------------------
+void vtkF3DQuakeMDLImporter::EnableAnimation(vtkIdType animationIndex)
+{
+  assert(animationIndex < static_cast<vtkIdType>(this->Internals->AnimationNames.size()));
+  assert(animationIndex >= 0);
+  this->Internals->EnableAnimation(animationIndex);
+}
+
+//----------------------------------------------------------------------------
+void vtkF3DQuakeMDLImporter::DisableAnimation(vtkIdType animationIndex)
+{
+  assert(animationIndex < static_cast<vtkIdType>(this->Internals->AnimationNames.size()));
+  assert(animationIndex >= 0);
+  this->Internals->DisableAnimation(animationIndex);
+}
+
+//----------------------------------------------------------------------------
+bool vtkF3DQuakeMDLImporter::IsAnimationEnabled(vtkIdType animationIndex)
+{
+  assert(animationIndex < static_cast<vtkIdType>(this->Internals->AnimationNames.size()));
+  assert(animationIndex >= 0);
+  return std::count(this->Internals->ActiveAnimationIds.begin(),
+           this->Internals->ActiveAnimationIds.end(), animationIndex) > 0;
+}
+
+//----------------------------------------------------------------------------
+bool vtkF3DQuakeMDLImporter::GetTemporalInformation(vtkIdType animationIndex, double frameRate,
+  int& nbTimeSteps, double timeRange[2], vtkDoubleArray* vtkNotUsed(timeSteps))
+{
+  this->Internals->SetFrameRate(frameRate);
```

Comment:
In F3D we dont care about the framerate. See `vtkF3DAssimpImporter::GetTemporalInformation`

Reply:
unless you need a framerate because its not defined by the model ?